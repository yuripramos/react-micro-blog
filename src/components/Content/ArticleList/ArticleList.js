import React from "react";
import { array } from "prop-types";
import Icon from "../../common/Icon";
import moment from "moment";
import {
  Title,
  Item,
  ListWrapper,
  Description,
  Info,
  InfoStyle,
  Name,
  Empty
} from "./styles";

function ArticleList({ articlesList }) {
  const isFilled = articlesList && articlesList.length > 0;
  return (
    <ListWrapper>
      <Title>Articles</Title>
      {isFilled ? (
        articlesList.map((e, i) => (
          <Item key={i}>
            <Info>
              <Description recipient>
                <Icon name="Papers" />
              </Description>
              <Name>{e.title}</Name>
              <InfoStyle>{`Published at:${moment(e.metadata.publishedAt).format(
                "L"
              )}`}</InfoStyle>
            </Info>
          </Item>
        ))
      ) : (
        <Empty>No Articles to display</Empty>
      )}
    </ListWrapper>
  );
}

export default ArticleList;

ArticleList.propTypes = {
  articlesList: array
};
