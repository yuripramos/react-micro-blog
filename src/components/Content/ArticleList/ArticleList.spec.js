import React from "react";
import { shallow } from "enzyme";
import ArticleList from "./ArticleList";

const props = {
  onFilter: jest.fn(),
  defaultFilter: {
    type: "author",
    range: 5
  }
};

describe("ArticleList component", () => {
  it("should match snapshot", () => {
    expect(shallow(<ArticleList {...props} />)).toMatchSnapshot();
  });
});
