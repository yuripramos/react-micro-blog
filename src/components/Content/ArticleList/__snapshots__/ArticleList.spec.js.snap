// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    defaultFilter={
      Object {
        "author": "name 1",
        "type": "author",
      }
    }
    onFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Articles
        </ForwardRef>,
        <ForwardRef>
          No Articles to display
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Articles",
        },
        "ref": null,
        "rendered": "Articles",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  font-size: ",
              "1rem",
              ";
  color: ",
              "#34b896",
              ";
  text-align: center;
  font-weight: 700
  width: 100%;
  padding: ",
              "0.625rem",
              ";
",
            ],
          },
          "displayName": "styled.span",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "No Articles to display",
        },
        "ref": null,
        "rendered": "No Articles to display",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  width: 100%;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  display: inline-flex;
  flex-wrap: wrap;
  background: ",
          "#efefef",
          ";
  color: ",
          "#333",
          ";
  height: 91vh;
  min-width: ",
          "12.5rem",
          ";
  ",
          "
    @media (min-width: ",
          "73.6875rem",
          ") {
      ",
          "
    width: ",
          "22.5rem",
          ";
  ",
          ";
    }
  ",
          ";
  align-content: flex-start;
  overflow-y: auto;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Articles
          </ForwardRef>,
          <ForwardRef>
            No Articles to display
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Articles",
          },
          "ref": null,
          "rendered": "Articles",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
  font-size: ",
                "1rem",
                ";
  color: ",
                "#34b896",
                ";
  text-align: center;
  font-weight: 700
  width: 100%;
  padding: ",
                "0.625rem",
                ";
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "No Articles to display",
          },
          "ref": null,
          "rendered": "No Articles to display",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  width: 100%;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  display: inline-flex;
  flex-wrap: wrap;
  background: ",
            "#efefef",
            ";
  color: ",
            "#333",
            ";
  height: 91vh;
  min-width: ",
            "12.5rem",
            ";
  ",
            "
    @media (min-width: ",
            "73.6875rem",
            ") {
      ",
            "
    width: ",
            "22.5rem",
            ";
  ",
            ";
    }
  ",
            ";
  align-content: flex-start;
  overflow-y: auto;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
